### PR TITLE
i guess i have to review voremob prs from now on because mimics just broke my goddamn linter and people need to remember that just because the last guy didn't code well doesn't mean there's an excuse to copy awful code

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 1]
```diff
@@ -342,51 +337,43 @@
...
-			visible_message("<font color='red'><b>The [new_mimic] suddenly growls beneath you as it turns out to be a mimic!</b></font>")
-			forceMove(new_mimic)
-			new_mimic.real_crate = src
-			new_mimic.name = name
-			new_mimic.desc = desc
-			new_mimic.icon = icon
-			new_mimic.icon_state = "wmimicopen"
-			new_mimic.icon_living = "wmimicopen"
-		else
-			qdel(src.loc)
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:369]: Y'SEE THIS?

DON'T.
DO.
THIS.

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 2]
```diff
@@ -342,51 +337,43 @@
 	icon_state = "wmimic"
 	var/mimic_chance = 30
 	var/mimic_active = TRUE
+	var/mimic_type = /mob/living/simple_mob/vore/aggressive/mimic/floor
 
-/obj/effect/floormimic/Crossed(AM as mob|obj)
+/obj/effect/floormimic/Crossed(atom/movable/AM)
 	. = ..()
-	Bumped(AM)
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:348]: OR THIS!!

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 3]
```diff
@@ -342,51 +337,43 @@
...
+	var/mob/living/simple_mob/vore/aggressive/mimic/floor/new_mimic = new mimic_type(drop_location())
+	visible_message("<span class='boldwarning'>The [new_mimic] suddenly growls beneath you as it turns out to be a mimic!</span>")
 
-/obj/effect/floormimic/attackby(obj/item/W as obj, mob/M as mob|obj)
+/obj/effect/floormimic/attackby(obj/item/I, mob/living/L)
 	if(mimic_active)
-		awaken(M)
+		awaken(L)
 	else
-		qdel(src.loc)
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:381]: OR THIS.

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 4]
```diff
@@ -342,51 +337,43 @@
...
-		qdel(src)
 		return ..()
 
-/obj/effect/floormimic/ex_act(severity) //Stores Mimic Contents for later
-	for(var/obj/O in src.contents)
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:386]: THERE SHOULD BE NOTHING IN CONTENTS.

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 5]
```diff
@@ -421,39 +409,15 @@
...
+	density = FALSE
+	anchored = TRUE
 	icon_state = "tmimic"
-
-/obj/effect/floormimic/tile/awaken(var/mob/living/M)
-	if(mimic_active)
-		mimic_active = FALSE
-		if(prob(mimic_chance))
-			var/mob/living/simple_mob/vore/aggressive/mimic/floor/new_mimic = new(loc, src)
-			visible_message("<font color='red'><b>The [new_mimic] suddenly growls beneath you as it turns out to be a mimic!</b></font>")
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:440]: CODE COPYPASTING IS BANNED

File: code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm [hunk 6]
```diff
@@ -342,51 +337,43 @@
...
+		return
+	var/mob/living/L = victim
+	if(L.hovering)
+		return
+	awaken(L)
+
+/obj/effect/floormimic/proc/awaken(mob/living/L)
+	if(!mimic_active)
 		qdel(src)
-		return ..()
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/vore/mimic.dm:375]: USE VISUAL STUDIO CODE, CHECK THE LINTER, IF IT SAYS YOU HAVE AN ERROR, FIX IT.

**Review by nik707 — APPROVED** [review]

**Pull request merged**